spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/configclient/ZookeeperConfigServerBootstrapper.java

```diff
@@ -67,27 +66,28 @@ public void initialize(BootstrapRegistry registry) {
 				ClassUtils.isPresent("org.springframework.cloud.bootstrap.marker.Marker", null)) {
 			return;
 		}
+
 		// create curator
-		CuratorFactory.registerCurator(registry, null, true, bootstrapContext -> isEnabled(bootstrapContext.get(Binder.class)));
+		CuratorFactory.registerCurator(registry, null, true, ZookeeperConfigServerBootstrapper::isEnabled);
 
 		// create discovery
 		registry.registerIfAbsent(ZookeeperDiscoveryProperties.class, context -> {
-			Binder binder = context.get(Binder.class);
-			if (!isEnabled(binder)) {
+			PropertyResolver propertyResolver = getPropertyResolver(context);
+			if (!isEnabled(context)) {
 				return null;
 			}
```

Comment:
```suggestion
			if (!isEnabled(context)) {
				return null;
			}
			PropertyResolver propertyResolver = getPropertyResolver(context);
```
The `getPropertyResolver` call could be moved down below the short-circuit block to avoid unnecessary work when we `return null` anyway.

Alternatively, we could reuse this `propertyResolver` for the `isEnabled` check if we added an overload of that method that accepts a `PropertyResolver` directly instead of extracting it from the `BootstrapContext` itself.